stl/inc/ranges

```diff
@@ -6821,6 +6822,496 @@ namespace ranges {
+                (noexcept(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>&>() -=
+                          static_cast<iter_difference_t<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>>(
+                              _STD declval<difference_type>()))
+                    && ...)) // strengthened
+                requires _All_random_access<_IsConst, _ViewTypes...> {
+                _Tuple_for_each(
+                    [&_Off]<class _IteratorType>(
+                        _IteratorType& _Itr) { _Itr -= static_cast<iter_difference_t<_IteratorType>>(_Off); },
+                    _Current);
+
+                return *this;
+            }
+
+            _NODISCARD constexpr auto operator[](const difference_type _Where) const
+                noexcept((noexcept((_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>())
+                                  [static_cast<iter_difference_t<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>>(
+                                      _STD declval<difference_type>())])
+                          && ...)) // strengthened
+                requires _All_random_access<_IsConst, _ViewTypes...> {
+                return _Tuple_transform(
+                    [&_Where]<class _IteratorType>(_IteratorType& _Itr) -> decltype(auto) {
+                        return _Itr[static_cast<iter_difference_t<_IteratorType>>(_Where)];
+                    },
+                    _Current);
+            }
+
+            _NODISCARD_FRIEND constexpr bool operator==(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+                noexcept(_STD declval<_Iterator_tuple_type>() == _STD declval<_Iterator_tuple_type>())) // strengthened
+                requires(equality_comparable<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {
+                if constexpr (!_Zip_is_common<_Maybe_const<_IsConst,
+                                  _ViewTypes>...> || _All_random_access<_IsConst, _ViewTypes...>) {
+                    return (_STD get<0>(_Lhs._Current) == _STD get<0>(_Rhs._Current));
+                } else if constexpr (_All_bidirectional<_IsConst, _ViewTypes...>) {
+                    return (_Lhs._Current == _Rhs._Current);
+                } else {
+                    return _Zip_evaluate_iterator_sentinel_equality(_Lhs._Current, _Rhs._Current);
+                }
+            }
+
+            _NODISCARD_FRIEND constexpr auto operator<=>(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+                noexcept(_STD declval<_Iterator_tuple_type>() <=> _STD declval<_Iterator_tuple_type>())) // strengthened
+                requires _All_random_access<_IsConst, _ViewTypes...> {
+                return (_Lhs._Current <=> _Rhs._Current);
+            }
+
+            _NODISCARD_FRIEND constexpr _Iterator operator+(const _Iterator& _Lhs, const difference_type _Rhs) noexcept(
+                noexcept(_STD declval<_Iterator>() += _Rhs)

```

Comment:
```suggestion
                noexcept(_STD declval<_Iterator&>() += _Rhs)
```

okay I know that there isn't a _real_ difference here in this specific case, but you are actually doing a `+=` on an lvalue. (same in `operator-`)